README.md

```diff
@@ -1,4 +1,41 @@
 # Data explorer
 
-## One-time setup
+## Architecture overview
+
+For local development, nginx is used to get around CORS.
+
+![Architecture overview](https://i.imgur.com/tKv8FEL.png)
+
+## Development
+
+### Updating the API using swagger-codegen
+We use [swagger-codegen](https://github.com/swagger-api/swagger-codegen) to
+automatically implement the API, as defined in `api/api.yaml`, for the API
+server and the UI. Whenever the API is updated, follow these steps to
+update the server implementations:
+
+TODO(melissachang): Add Javascript definitions for React.
+
+1. If you do not already have the jar, you can download it here:
+  ```
+  # Linux
+  wget http://central.maven.org/maven2/io/swagger/swagger-codegen-cli/2.2.3/swagger-codegen-cli-2.2.3.jar -O swagger-codegen-cli.jar
+  # macOS
+  brew install swagger-codegen
+  ```
+1. Clear out existing generated models:
```

Comment:
These aren't rendering correctly in github unfortunately. I think there's a bug that when numbered lists contain code blocks they don't increment automatically. Mind just switch to `1,2,3...`?

Reply:
Realized you probably took part of this from the JM readme which has the same issue: https://github.com/DataBiosphere/job-manager#updating-the-api-using-swagger-codegen